asyncscope.md

```diff
@@ -279,201 +352,541 @@ listening activity is bound in the scope of the loop, the lifetime of handling r
 ```
 
-Returns a _`nest-sender`_ that, when started, adds the given sender to the count of senders that the `counting_scope`
-object will require to complete before it will destruct.
+Atomically increments the scope's count of outstanding senders if and only if the scope is in the open state and then
+returns a _`nest-sender`_.
+
+If the atomic increment succeeded then the return value will be an associated _`nest-sender`_ that contains a reference
+to `this` (the associated scope) and a copy of the input sender, `s`, that is copy- or move-constructed from `s`. If
+this copy or move throws then, before the exception is allowed to escape to the caller, the atomic increment needs to be
+undone with a decrement to provide the strong exception guarantee.
 
-A call to `nest()` does not start the given sender. A call to `nest()` is not expected to incur allocations.
+If the atomic increment failed then the return value will be an unassociated _`nest-sender`_ and no exceptions are
+possible. In this case, the return value does not store a reference to `this` and the given sender, `s`, is discarded.
 
-The sender returned by a call to `nest()` holds a reference to the `counting_scope` in order to add the given sender to
-the count of senders when it is started. Connecting and starting the sender returned from `nest()` will connect and
-start the given sender and add the given sender to the count of senders that the `counting_scope` object will require to
-complete before it will destruct.
+An associated _`nest-sender`_ is a kind of RAII handle to the scope; it is responsible for decrementing the scope's
+count of outstanding senders in its destructor unless that responsibility is first given to some other object.
+Move-construction and move-assignment transfer the decrement responsibility to the destination instance. Connecting an
+instance to a receiver transfers the decrement responsibility to the resulting _`operation-state`_, which must meet the
+responsibility when the operation completes or is destroyed, whichever comes first (note: if the _`operation-state`_
+is started, then the decrement should happen *after* invoking a completion method on its receiver to ensure that any
+reference produced by the nested sender is not dangling at the time of invocation). Whenever the balancing decrement
+happens (including if it happens as a side effect of allowing an exception to escape from `nest()`), it's possible that
+the scope has transitioned to the closed/joining state since the _`nest-sender`_ was constructed, which means that there
+is a _`join-sender`_ waiting to complete so, if the decrement brings the count of outstanding senders to zero then the
+waiting _`join-sender`_ needs to be notified that the scope is now joined and the sender can complete.
+
+A call to `nest()` does not start the given sender. A call to `nest()` is not expected to incur allocations other than
+whatever might be required to move or copy `s`.
```

Comment:
can a `nest-sender` be multi-shot? what would that do to the scope's count? If the answer is that `nest-sender` is single-shot, then it should be move-only.